CHANGELOG.develop

```diff
@@ -299,6 +301,13 @@ sane way, here is the complet list of changed key bindings:
   - init function of all packages
   - post-init functions for all packages  (thanks to Sylvain Benner)
 - Backported battery fix for macOS (thanks to Sylvain Benner)
+- Defer packages by default using =use-package-always-defer= (thanks to Sylvain
+  Benner)
+- Regressions caused by removing =:defer t=. Fixed it by adding =:demand t=.
```

Comment:
Btw, I added a "various regression fixes" section at the bottom in my previous PR (not yet merged):
https://github.com/syl20bnr/spacemacs/pull/11772/files#diff-f3337695f47c8cf4b3d417aa7971aa6bR1144
 
I think makes more sense for non user-facing regression fixes like this to go there? Also since they're quite hard to describe

Reply:
So, they should be removed?

Reply:
I was thinking you could move it to the same place under the Various improvements section:

```
- Various regression fixes (thanks to bmag)
```

so it'll be merged accordingly :)

Reply:
I believe the following guideline applies:

> If you notice that a commit builds on top of an existing one you can group those together.

Sylvain made the change to defer by default, and bmag built on that change by adding some necessary overrides.

Where this gets confusing is that Sylvain subsequently reverted the change, changing the default for `use-package-always-defer` back to `nil`, which means that these two change log entries may ultimately be unnecessary.  In light of that, and since Sylvain and and bmag are already listed under "Various code style improvements", I believe that these two entries related to deferred loading can be omitted entirely.

Reply:
I am also curious about the lazy loading things. They are done all over the history IMO. So, should they be put in a specific section?